### PR TITLE
Of11 amr path logic

### PR DESCRIPTION
**Pull request by colemanjs**
Updated uniformIntervals to set the interval time based on the total scanning time. Also, improved the stability of the thermal solver by removing the explicit latent heat correction.

### CODE REVIEW

File: applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C [hunk 1]
```diff
@@ -49,153 +54,148 @@ Foam::refinementControllers::uniformIntervals::uniformIntervals
...
+                cellBb.min()
+                    = min(cellBb.min(), points[vertices[j]] - extend);
+                cellBb.max()
+                    = max(cellBb.max(), points[vertices[j]] + extend);
+            }
+
+            cellBbs[celli] = cellBb;
+        }
+
+        //- Update refinement marker field:
```
Comment (kincaidkc) [inline applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C:134]: can we use a consistent comment format? I've been using:

```
//- comment line 1
//  comment line 2
//  blah blah blah
```

sometimes you use `//-`, sometimes just `//`. i don't particularly care which we settle one, just that it's consistent

File: applications/solvers/additiveFoam/thermo/TEqn.H [hunk 2]
```diff
@@ -69,4 +69,13 @@
             break;
         }
     }
+
+    // set solid fraction consistent with temperature
```
Comment (kincaidkc) [inline applications/solvers/additiveFoam/thermo/TEqn.H:73]: this is what fixed the issues during coarsening?

File: tutorials/AMB2018-01-L7/system/decomposeParDict [hunk 3]
```diff
@@ -14,7 +14,7 @@ FoamFile
 }
 // * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
 
-numberOfSubdomains 4;
+numberOfSubdomains 16;
```
Comment (kincaidkc) [inline tutorials/AMB2018-01-L7/system/decomposeParDict:17]: do we want to set this lower so people that download can still run on their laptops? 9 would probably work for most people these days and is still a perfect square

File: tutorials/AMB2018-02-B/system/fvSolution [hunk 4]
```diff
@@ -31,6 +31,14 @@ solvers
         relTol          0;
     }
 
+    "pcorr.*"
```
Comment (kincaidkc) [inline tutorials/AMB2018-02-B/system/fvSolution:34]: why did you have to add a `pcorr` solver?

File: applications/solvers/additiveFoam/movingHeatSource/movingHeatSourceModel/movingHeatSourceModel.C [hunk 5]
```diff
@@ -151,6 +151,11 @@ void Foam::movingHeatSourceModel::update()
             qDot_ += qDoti;
         }
     }
+
+    qDot_.correctBoundaryConditions();
+
+    // update the heat source-based AMR field
+    refinementController_->update();
```
Comment (kincaidkc) [inline applications/solvers/additiveFoam/movingHeatSource/movingHeatSourceModel/movingHeatSourceModel.C:158]: nice, glad we were able to hide this from the main solver

**Review by kincaidkc — COMMENTED** [review]
first pass, some minor changes and comments

File: applications/solvers/additiveFoam/movingHeatSource/movingBeam/movingBeam.C [hunk 6]
```diff
@@ -58,19 +58,18 @@ Foam::movingBeam::movingBeam
     deltaT_(GREAT),
     hitPathIntervals_(true)
 {
-    //- Get beam parameters
-    deltaT_ = beamDict_.lookupOrDefault<scalar>("deltaT", GREAT);
-    hitPathIntervals_ = beamDict_.lookupOrDefault<bool>("hitPathIntervals", true);
-    
-    //- Read scan path file
+    deltaT_ =
```
[the remark this answers is not in the document]
Comment (colemanjs) [reply applications/solvers/additiveFoam/movingHeatSource/movingBeam/movingBeam.C:61]: I am doing this manually whenever I notice code formatting that breaks the style guide (https://openfoam.org/dev/coding-style-guide/). We should probably do a PR in the near future where we review our code for formatting errors

[the remark this answers is not in the document]
Comment (kincaidkc) [reply applications/solvers/additiveFoam/movingHeatSource/movingBeam/movingBeam.C:61]: it looks like there is active discussion about a clang formatter for OpenFOAM. we should watch this thread and consider implementing whatever solution they arrive on.

https://develop.openfoam.com/Development/openfoam/-/issues/1634

File: applications/solvers/additiveFoam/thermo/TEqn.H (hunk 2 above)
Comment (colemanjs) [reply applications/solvers/additiveFoam/thermo/TEqn.H:73]: But this also means that alpha1 is not forced to 1 when T < Tsol, or 0 when T > Tliq, so we must apply this explicit correction.

File: applications/solvers/additiveFoam/movingHeatSource/movingBeam/movingBeam.C (hunk 6 above)
[the remark this answers is not in the document]
Comment (colemanjs) [reply applications/solvers/additiveFoam/movingHeatSource/movingBeam/movingBeam.C:61]: Agreed

File: applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C (hunk 1 above)
Comment (colemanjs) [reply applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C:134]: Agreed. I tried to be more consistent

File: tutorials/AMB2018-01-L7/system/decomposeParDict (hunk 3 above)
Comment (colemanjs) [reply tutorials/AMB2018-01-L7/system/decomposeParDict]: Yes - reverted

**Pull request merged**